nixos/modules/services/web-apps/stirling-pdf.nix

```
@@ -0,0 +1,140 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    getExe
+    literalExpression
+    mkEnableOption
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    optional
+    ;
+
+  inherit (lib.types)
+    attrsOf
+    nullOr
+    path
+    port
+    str
+    submodule
+    ;
+
+  cfg = config.services.stirling-pdf;
+in
+
+{
+  meta.maintainers = with lib.maintainers; [ thubrecht ];
+
+  options.services.stirling-pdf = {
+    enable = mkEnableOption "Stirling-PDF, a web application for manipulating PDF files";
+
+    package = mkPackageOption pkgs "stirling-pdf" { };
+
+    domain = mkOption {
+      type = str;
+      description = ''
+        The domain for the application.
+      '';
+    };
+
+    port = mkOption {
+      type = port;
+      default = 8080;
+      description = ''
+        Port that Stirling-PDF will listen on.
+      '';
+    };
+
+    environment = mkOption {
+      type = attrsOf str;
+      defaultText = ''
+        {
+          SERVER_PORT = cfg.port;
+        }
+      '';
+      description = ''
+        Settings for the stirling-pdf app.
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = nullOr path;
+      default = null;
+      description = ''
+        File containing additional environment variables to pass to Stirling PDF.
+      '';
+    };
+
+    nginx = mkOption {
+      type = nullOr (submodule (import ../web-servers/nginx/vhost-options.nix { inherit config lib; }));
+      default = null;
+      description = ''
+        With this option, you can customize the nginx virtualHost settings.
+      '';
+      example = literalExpression ''
+        {
+          # To enable encryption and let Let's Encrypt take care of certificate
+          forceSSL = true;
+          enableACME = true;
+        }
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.stirling-pdf = {
+      inherit (cfg) environment;
+
+      path = with pkgs; [
+        calibre
+        libreoffice
+        ocrmypdf
+        opencv
+        pngquant
+        tesseract
+        unoconv
+        unpaper
+
+        python3.pkgs.weasyprint
+      ];
+
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        BindReadOnlyPaths = [ "${pkgs.tesseract}/share/tessdata:/usr/share/tessdata" ];
+        DynamicUser = true;
+        CacheDirectory = "stirling-pdf";
+        Environment = [ "HOME=%S/stirling-pdf" ];
+        EnvironmentFile = optional (cfg.environmentFile != null) cfg.environmentFile;
+        ExecStart = getExe cfg.package;
+        RuntimeDirectory = "stirling-pdf";
+        StateDirectory = "stirling-pdf";
+        SuccessExitStatus = 143;
+        User = "stirling-pdf";
+        WorkingDirectory = "/var/lib/stirling-pdf";
+      };
+    };
+
+    services.stirling-pdf.environment = {
+      # Default values
+      SERVER_PORT = builtins.toString cfg.port;
+      INSTALL_BOOK_AND_ADVANCED_HTML_OPS = "true";
```

Would the user want to override this?
```suggestion
      INSTALL_BOOK_AND_ADVANCED_HTML_OPS = lib.mkDefault "true";
```

---

Hmm, not really, it is disabled by default upstream because you need to install calibre I think